typescript/infra/scripts/utils.ts

```diff
@@ -196,6 +204,43 @@ export async function getMultiProviderForRole(
   return multiProvider;
 }
 
+export async function getKeysForRole(
+  txConfigs: ChainMap<ChainMetadata>,
+  environment: DeployEnvironment,
+  context: Contexts,
+  role: Role,
+  index?: number,
+): Promise<ChainMap<string>> {
+  if (process.env.CI === 'true') {
+    return {};
+  }
+
+  const keys = await promiseObjAll(
+    objMap(txConfigs, async (chain, _) => {
+      const key = getKeyForRole(environment, context, chain, role, index);
+      return key.privateKey;
+    }),
+  );
+  return keys;
+}
+
+export function getAddressesForKey(
+  keys: ChainMap<string>,
+  chain: ChainName,
+  manager: ChainMetadataManager<any>,
+) {
+  const protocol = manager.getChainMetadata(chain).protocol;
+  if (protocol === ProtocolType.Ethereum) {
+    return new Wallet(keys[chain]).address;
+  } else if (protocol === ProtocolType.Sealevel) {
+    return Keypair.fromSecretKey(
+      base58ToBuffer(keys[chain]),
```

Comment:
flagging this should probably be hex, similar to what's done in `typescript/infra/src/agents/gcp.ts`. Probably is an opportunity to dedupe this too given the per-protocol logic will be exactly the same